### ISSUE TITLE
Consider using GDB MI commands

### ISSUE THREAD

**Issue by lebr0nli** [lebr0nli/GEP#41]
`gdb.execute_mi` is added after GDB 14.1

GDB MI `-complete` command somehow can provide better competitions then normal `complete` command:
```python
>>> gdb.execute_mi("-complete","enable ")
{'completion': 'enable ', 'matches': ['enable breakpoints', 'enable count', 'enable delete', 'enable display', 'enable frame-filter', 'enable mem', 'enable once', 'enable pretty-printer', 'enable probes', 'enable type-printer', 'enable unwinder', 'enable xmethod'], 'max_completions_reached': '0'}
>>> gdb.execute("complete enable ")
enable
```

**Comment by lebr0nli** [comment]
Wow, the performance seems also better:
```python
>>> timeit.timeit("import gdb;gdb.execute('complete',to_string=True)", number=1000)
2.3764819998759776
>>> timeit.timeit("import gdb;gdb.execute_mi('-complete','')",number=1000)
1.2388369999825954
```

**Comment by lebr0nli** [comment]
Should be resolved in #42
